chalk-engine/src/simplify.rs

```diff
@@ -68,14 +69,22 @@ impl<I: Interner, C: Context<I>> Forest<I, C> {
                     &goal.b,
                     &mut ex_clause,
                 )?,
-                GoalData::DomainGoal(domain_goal) => {
-                    ex_clause
-                        .subgoals
-                        .push(Literal::Positive(InEnvironment::new(
+                GoalData::DomainGoal(domain_goal) => match domain_goal {
```

Comment:
it seems like this logic is missing from the recursive solver?

Reply:
I think you would add it here

https://github.com/rust-lang/chalk/blob/ffa2e515a4fac85f053dcbe6307d73ebc0e82043/chalk-solve/src/recursive/fulfill.rs#L174-L180

but it'd be nice to refactor the two solvers to share this logic.